sdk/openai/Azure.AI.OpenAI/src/Custom/CompletionsOptions.cs

```diff
@@ -147,26 +161,31 @@ public partial class CompletionsOptions
         public IDictionary<int, int> TokenSelectionBiases { get; }
 
         internal bool? InternalShouldStreamResponse { get; set; }
-        internal string InternalNonAzureModelName { get; set; }
+
+        internal IDictionary<string, int> InternalStringKeyedTokenSelectionBiases { get; }
 
         /// <summary> Initializes a new instance of CompletionsOptions. </summary>
+        /// <param name="deploymentName"> The deployment name to use for this request. </param>
         /// <param name="prompts"> The prompts to generate completions from. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="prompts"/> is null. </exception>
-        public CompletionsOptions(IEnumerable<string> prompts)
+        public CompletionsOptions(string deploymentName, IEnumerable<string> prompts)
```

Comment:
Great point! I think we originally made `model` optional in TypeSpec since it was superseded by the method parameter, but you're right that this is now effectively realigning it and making it required again -- that warrants an update to the TypeSpec once all languages are aligned. Doesn't need to block this PR, but this PR should reflect the principle.

I agree on the consistency; every options should have:

1. A default constructor allows init-style assignment
2. (As long as there are N or fewer) a constructor that accepts *all* truly required parameters

That means a few other Options classes need the same new constructor treament -- and it also means that a few need to hide the now-"in-between" constructors that only have "some" required parameters (like here, the one with just `IEnumerable<string> prompts` that still needs you to set `DeploymentName`)

I'll update that and double-check all the exported constructors.